### DISCUSSION TITLE
Processing metadata -> Failed (no information)

### DISCUSSION THREAD

**Discussion by aaronjamt** [tianocore/edk2#3666] [Q&A] [answered]
I'm trying to compile [this project](https://github.com/DHowett/FrameworkHacksPkg) but, after cloning EDK2 (with `--recurse-submodules`), cloning the project into the EDK2 directory, running `source edksetup.sh`, and trying to `build`, all I get is the following:
```
Build environment: Linux-6.0.0-kali3-amd64-x86_64-with-glibc2.36
Build start time: 17:17:22, Nov.27 2022

WORKSPACE        = .../edk2
EDK_TOOLS_PATH   = .../edk2/BaseTools
CONF_PATH        = .../edk2/Conf
PYTHON_COMMAND   = /usr/bin/python3


Processing meta-data 
Architecture(s)  = X64
Build target     = RELEASE
Toolchain        = VS2015x86

Active Platform          = .../edk2/FrameworkHacksPkg/FrameworkHacksPkg.dsc
... 

- Failed -
Build end time: 17:17:25, Nov.27 2022
Build total time: 00:00:03
```

There is no information about what's wrong and the closest discussion to this issue I could find (#3629) was on Windows using Python 2 (but I'm using Kali Linux with Python 3.10). Can anyone help?

Edit: I also tried to build https://github.com/lersi/edk2_project_template (linked in #3629) and had the same issue.

**Comment by lersi** [comment]
first of all, sorry that my project did not help :(

have you built basetools, before running `build` command?
you can compile basetools by using the command `make -C edk2/BaseTools`

you can contact me trough my email, for discussing this issue farther.

**Reply by lersi** [reply]
I have tried to compile this project using my build system, and it currently works on linux, as it can be seen by this [workflow run](https://github.com/lersi/Compile-FrameworkHacksPkg/actions/runs/3565363184) (release 0.0.4)

**Reply by aaronjamt** [reply]
I just ran `make -C BaseTools` from the `edk2` directory and it definitely compiled some stuff but I ran `build` again and got the exact same error.

**Comment by ajfish** [comment]
Pro Tip: If you want prompt help copy the error message into the radar so people don’t have to invest lots of time tracking down the information…..

Mac Build Failure:
Undefined symbols for architecture x86_64:
55
 <https://github.com/lersi/Compile-FrameworkHacksPkg/actions/runs/3565363184/jobs/5990425850#step:8:56>  "_memset", referenced from:
56
 <https://github.com/lersi/Compile-FrameworkHacksPkg/actions/runs/3565363184/jobs/5990425850#step:8:57>
57
 <https://github.com/lersi/Compile-FrameworkHacksPkg/actions/runs/3565363184/jobs/5990425850#step:8:58>      _cmd_version in ECTool.lib(ECTool.obj)
58
 <https://github.com/lersi/Compile-FrameworkHacksPkg/actions/runs/3565363184/jobs/5990425850#step:8:59>build.py...
59
 <https://github.com/lersi/Compile-FrameworkHacksPkg/actions/runs/3565363184/jobs/5990425850#step:8:60>      _cmd_console in ECTool.lib(ECTool.obj)
60
 <https://github.com/lersi/Compile-FrameworkHacksPkg/actions/runs/3565363184/jobs/5990425850#step:8:61> : error 7000: Failed to execute command
61
 <https://github.com/lersi/Compile-FrameworkHacksPkg/actions/runs/3565363184/jobs/5990425850#step:8:62>      _cmd_reflash in ECTool.lib(FWUpdate.obj)
62
 <https://github.com/lersi/Compile-FrameworkHacksPkg/actions/runs/3565363184/jobs/5990425850#step:8:63>	make tbuild [/Users/runner/work/Compile-FrameworkHacksPkg/Compile-FrameworkHacksPkg/build/Build/FrameworkHacksPkg/RELEASE_XCODE5/X64/FrameworkHacksPkg/Application/ECTool/ECTool]
63
 <https://github.com/lersi/Compile-FrameworkHacksPkg/actions/runs/3565363184/jobs/5990425850#step:8:64>      _flash_read in ECTool.lib(Flash.obj)
64
 <https://github.com/lersi/Compile-FrameworkHacksPkg/actions/runs/3565363184/jobs/5990425850#step:8:65>      _flash_write in ECTool.lib(Flash.obj)
65
 <https://github.com/lersi/Compile-FrameworkHacksPkg/actions/runs/3565363184/jobs/5990425850#step:8:66>A memset link failure is usually assigning to zero a variable bigger thank 64-bits. If you compiled with optimizations disable (-O0 compiler flag) you always fail like this. With the normal size optimization (-Os) it is possible that the optimizer will optimize away the memset and the code will compile. This exposes you do different behavior between different compilers, and even different compiler versions. Generally how I’d debug this is to:
Look at the assembler to see the call to memset and then track that back to the code. 
$ lldb  ECTool.obj
(lldb) dis -m -n cmd_version

But it is likely just some local variable (like a struct) getting set to zero in the cmd_console() command. 


Windows error:
D:\a\Compile-FrameworkHacksPkg\Compile-FrameworkHacksPkg\uefi_apps\FrameworkHacksPkg\Include\Library/CrosECLib.h(597): error C2143: syntax error: missing ')' before '(' [D:\a\Compile-FrameworkHacksPkg\Compile-FrameworkHacksPkg\build\FrameworkHacksPkg.vcxproj]
45
 <https://github.com/lersi/Compile-FrameworkHacksPkg/actions/runs/3565363184/jobs/5990425635#step:6:46>This looks like a bug in some include file?


Thanks,

Andrew Fish

**Reply by aaronjamt** [reply]
There is no error information... that's why I'm asking here. Initially I had some errors and I was able to resolve them myself (well, with Google) but there is 0 information about what's wrong. Is there some debugging/verbose flag(s)? I tried `--verbose` and it adds some information but nothing resembling an error.

**Reply by lersi** [reply]
thank you for the help, my solution was a quick and hasty work and the propose was to show it can be compiled on Linux (and easily).

**Comment by aaronjamt** [comment, marked as answer]
Turns out I had to add `-t GCC5` to build with GCC5 as the toolchain instead of VS2017.
